### PR TITLE
runaway: change `tidb_runaway_watch` time type to UTC

### LINKED ISSUE #54770: runaway: change tidb_runaway_watch time type to UTC
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Currently `tidb_runaway_watch` relies on struct `QuarantineRecord` to store time as UTC, while `runaway_watch` directly selects `tidb_runaway_watch` to convert to local time.

```bash
mysql> SELECT * FROM mysql.tidb_runaway_watch limit 1\G
*************************** 1. row ***************************
                 id: 1
resource_group_name: default
         start_time: 2024-07-16 07:34:14.137009
           end_time: 2024-07-16 07:44:14.137009
              watch: 1
         watch_text: select  count(*) from orders where o_orderdate = '1994-11-12'
             source: 127.0.0.1:4000
             action: 3
1 row in set (0.00 sec)

mysql> SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES where id =1\G
*************************** 1. row ***************************
                 ID: 1
RESOURCE_GROUP_NAME: default
         START_TIME: 2024-07-16 15:34:14
           END_TIME: 2024-07-16 15:44:14
              WATCH: Exact
         WATCH_TEXT: select  count(*) from orders where o_orderdate = '1994-11-12'
             SOURCE: 127.0.0.1:4000
             ACTION: Kill
1 row in set (0.01 sec)
```

### 2. What did you expect to see? (Required)
`tidb_runaway_watch` displays UTC as well.

### 3. What did you see instead (Required)
`tidb_runaway_watch` displays local time while `tidb_runaway_watch` displays UTC

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #54770, Ref #54434

Problem Summary: runaway: change `tidb_runaway_watch` to UTC

### What changed and how does it work?

Currently `tidb_runaway_watch` relies on struct `QuarantineRecord` to store time as UTC, while `runaway_watch` directly selects `tidb_runaway_watch` to convert to local time. Let's change `tidb_runaway_watch` to UTC as well.
> **why not change `tidb_runaway_watch` to local time?**
because `tidb_runaway_watch` is `datetime`, which is different from `timestamp`, and can not convert UTC to local time automatically.
https://github.com/pingcap/tidb/blob/e5a8a23e836a99b95db201f412cb7fdd9c811c5f/pkg/session/bootstrap.go#L635-L646
This means the timezone information will be lost, and the user will not know which timezone to switch to when in crossing time zones (the user needs to confirm the local time zone on the server side).

before
```bash
mysql> SELECT * FROM mysql.tidb_runaway_watch limit 1\G
*************************** 1. row ***************************
                 id: 1
resource_group_name: default
         start_time: 2024-07-16 07:34:14.137009
           end_time: 2024-07-16 07:44:14.137009
              watch: 1
         watch_text: select  count(*) from orders where o_orderdate = '1994-11-12'
             source: 127.0.0.1:4000
             action: 3
1 row in set (0.00 sec)

mysql> SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES where id =1\G
*************************** 1. row ***************************
                 ID: 1
RESOURCE_GROUP_NAME: default
         START_TIME: 2024-07-16 15:34:14
           END_TIME: 2024-07-16 15:44:14
              WATCH: Exact
         WATCH_TEXT: select  count(*) from orders where o_orderdate = '1994-11-12'
             SOURCE: 127.0.0.1:4000
             ACTION: Kill
1 row in set (0.01 sec)
```

After
```bash
mysql> SELECT * FROM mysql.tidb_runaway_watch limit 1\G
*************************** 1. row ***************************
                 id: 1
resource_group_name: default
         start_time: 2024-07-19 07:18:36.572748
           end_time: 2024-07-19 07:28:36.572748
              watch: 1
         watch_text: select  count(*) from orders where o_orderdate = '1992-07-03'
             source: 127.0.0.1:4000
             action: 3
1 row in set (0.00 sec)

mysql> SELECT * FROM INFORMATION_SCHEMA.RUNAWAY_WATCHES where id =1\G
*************************** 1. row ***************************
                 ID: 1
RESOURCE_GROUP_NAME: default
         START_TIME: 2024-07-19 07:18:36
           END_TIME: 2024-07-19 07:28:36
              WATCH: Exact
         WATCH_TEXT: select  count(*) from orders where o_orderdate = '1992-07-03'
             SOURCE: 127.0.0.1:4000
             ACTION: Kill
1 row in set (0.00 sec)
```

### Check List

Tests 

- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
runaway: change tidb_runaway_watch time type to UTC
```

### CODE REVIEW

**Comment by HuSharp** [comment]
@nolouch @glorv @JmPotato  PTAL, thx!

File: pkg/executor/infoschema_reader.go [hunk 1]
```diff
@@ -3466,8 +3466,8 @@ func (e *memtableRetriever) setDataFromRunawayWatches(sctx sessionctx.Context) e
 		row := types.MakeDatums(
 			watch.ID,
 			watch.ResourceGroupName,
-			watch.StartTime.Local().Format(time.DateTime),
-			watch.EndTime.Local().Format(time.DateTime),
+			watch.StartTime.Format(time.DateTime),
+			watch.EndTime.Format(time.DateTime),
```
Comment (glorv) [inline pkg/executor/infoschema_reader.go:3470]:
```suggestion
			watch.StartTime.UTC().Format(time.DateTime),
			watch.EndTime.UTC().Format(time.DateTime),
```
Better always ensure the value is formatted under the UTC timezone.

**Review by glorv — APPROVED** [review]

**Comment by glorv** [comment]
BTW, I think we better cherry-pick this PR to v8.1 and 7.5 as it's more like a bug than enhancement.

**Comment by HuSharp** [comment]
> BTW, I think we better cherry-pick this PR to v8.1 and 7.5 as it's more like a bug than enhancement.

I created an issue https://github.com/pingcap/tidb/issues/54770, please help me add label to cp :)

**Comment by glorv** [comment]
> > BTW, I think we better cherry-pick this PR to v8.1 and 7.5 as it's more like a bug than enhancement.
> 
> I created an issue #54770, please help me add label to cp :)

Done

**Review by JmPotato — APPROVED** [review]

**Comment by HuSharp** [comment]
@JmPotato Can you help to add an `approved` label? :)

**Comment by JmPotato** [comment]
> @JmPotato Can you help to add an `approved` label? :)

You have to ask for an approval from the owners of `domain` pkg: https://github.com/pingcap/tidb/blob/master/pkg/domain/OWNERS

**Comment by HuSharp** [comment]
> > @JmPotato Can you help to add an `approved` label? :)
> 
> You have to ask for an approval from the owners of `domain` pkg: [`master`/pkg/domain/OWNERS](https://github.com/pingcap/tidb/blob/master/pkg/domain/OWNERS?rgh-link-date=2024-07-20T12%3A43%3A00Z)

got it.

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**